zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/Notebook.java

```diff
@@ -327,10 +327,9 @@ public String cloneNote(String sourceNoteId, String newNotePath, AuthenticationI
             newNote.setConfig(new HashMap<>(sourceNote.getConfig()));
             newNote.setInfo(new HashMap<>(sourceNote.getInfo()));
             newNote.setDefaultInterpreterGroup(sourceNote.getDefaultInterpreterGroup());
-            newNote.setNoteForms(new HashMap<>(sourceNote.getNoteForms()));
-            newNote.setNoteParams(new HashMap<>(sourceNote.getNoteParams()));
```

Comment:
I like this approach because it makes it clear that we are creating a flat copy of the HashMap. Your new approach does not show this.
Perhaps we should have `getNoteForms()` and `getNoteParams()` return an unmodifiable HashMap.

Reply:
You are right, I have reverted this change